torch_xla/experimental/spmd_fully_sharded_data_parallel.py

```diff
@@ -113,7 +118,8 @@ def __init__(
     for param in module.parameters():
       if torch_xla._XLAC._get_xla_sharding_spec(param) != "":
         continue
-      spmd.mark_sharding(param, mesh, _prepare_spmd_partition_spec(param))
+      spmd.mark_sharding(
+          param, mesh, _prepare_spmd_partition_spec(param, shard_maximal=True))
```

Comment:
should we make it configureable?

Reply:
No, not necessary... It shouldn't matter to the user...

Reply:
ok I think I get it.. We need to do a all-gather anyway before entering the layer and only one dimension is being sharded.

Reply:
Yea, that's right. If the 0th dim is like size of 8 (MoE) and we are sharding it on v5p-2048, it will be a disaster.